proxy/ProxyTransaction.h

```diff
@@ -114,15 +121,17 @@ class ProxyTransaction : public VConnection
   // This function must return a non-negative number that is different for two in-progress transactions with the same proxy_ssn
   // session.
   //
-  void set_rx_error_code(ProxyError e);
-  void set_tx_error_code(ProxyError e);
+  virtual void set_rx_error_code(ProxyError e);
+  virtual void set_tx_error_code(ProxyError e);
 
   bool support_sni() const;
 
   /// Variables
   //
   HttpSessionAccept::Options upstream_outbound_options; // overwritable copy of options
 
+  IOBufferReader *get_reader();
```

Comment:
It's almost bike-shedding and this is not the only place, but I don't like this name very much because the data source (or the direction) is unclear. My understanding is that this reader always reads data from a peer (i.e. a remote host) regardless of the type of transaction (ClientTransaction or ServerTransaction).

How about `get_remote_data_reader` ?

Reply:
I'm ok with get_remote_reader or remote_data_reader.  Just added this as a wrapper method around the _reader member.